src/dom/document.ts

```diff
@@ -307,8 +307,25 @@ export class Document extends Node {
   private _getElementsByClassName(className: string, search: Node[]): Node[] {
     for (const child of this.childNodes) {
       if (child.nodeType === NodeType.ELEMENT_NODE) {
-        if ((<Element> child).classList.contains(className)) {
-          search.push(child);
+        // remove duplicate class names
+        const classList = new Set(className.split(" "));
+
+        let matches: Node[] = [];
+        let matchesCount = 0;
+
+        for (const singleClassName of classList) {
+          if ((<Element> child).classList.contains(singleClassName)) {
+            matchesCount++;
+            // avoid pushing the same element multiple times due to multiple class names
+            if (!matches.includes(child)) {
+              matches.push(child);
+            }
+          }
+        }
+
+        // ensure that all class names are present
+        if (matchesCount >= classList.size) {
```

Comment:
After making the changes above, this should be changed to `=== classList.length`